@muthutt In this PR when I tried to run `test_outer.py` it failed with the following error. Hence that needs to be fixed.
`  def impl(tensor, device, *, memory_config):
        ttl_tensor = tensor.value
>       return Tensor(ttl_tensor.to(device, memory_config))
E       RuntimeError: TT_FATAL @ tt_metal/impl/buffers/buffer.cpp:30: page_size % sizeof(uint32_t) == 0
E       info:
E       Page size must be divisible by sizeof(uint32_t) because buffers hold uint32_t values
E       backtrace:
E        --- tt::tt_metal::Buffer::Buffer(tt::tt_metal::Device*, unsigned long, unsigned long, tt::tt_metal::BufferType, tt::tt_metal::TensorMemoryLayout, std::optional<tt::tt_metal::ShardSpecBuffer>)
E        --- tt::tt_metal::tensor_impl::detail::allocate_interleaved_buffer_on_device(unsigned int, tt::tt_metal::Device*, tt::tt_metal::Shape const&, tt::tt_metal::DataType, tt::tt_metal::Layout, tt::tt_metal::MemoryConfig const&)
E        --- tt::tt_metal::tensor_impl::allocate_buffer_on_device(unsigned int, tt::tt_metal::Device*, tt::tt_metal::Shape const&, tt::tt_metal::DataType, tt::tt_metal::Layout, tt::tt_metal::MemoryConfig const&, std::optional<tt::tt_metal::ShardSpecBuffer>)
E        --- /home/ubuntu/uma/tt-metal/tt_eager/tt_lib/_C.so(+0x580e78) [0x7f88f90cce78]
E        --- /home/ubuntu/uma/tt-metal/tt_eager/tt_lib/_C.so(_ZNSt8__detail9__variant17__gen_vtable_implILb1ENS0_12_Multi_arrayIPFSt10shared_ptrIN2tt8tt_metal6BufferEEOZNS5_11tensor_impl16to_device_bufferI8bfloat16EES7_RKSt7variantIJNS5_12OwnedStorageENS5_13DeviceStorageENS5_15BorrowedStorageEEEPNS5_6DeviceERKNS5_5ShapeENS5_8DataTypeENS5_6LayoutERKNS5_12MemoryConfigESt8optionalINS5_9ShardSpecEEEUlOT_E_SH_EJEEESt5tupleIJSH_EESt16integer_sequenceImJLm2EEEE14__visit_invokeESY_SH_+0xe4) [0x7f88f90d0014]
E        --- /home/ubuntu/uma/tt-metal/tt_eager/tt_lib/_C.so(+0x578d0d) [0x7f88f90c4d0d]
E        --- std::_Function_handler<tt::tt_metal::Tensor (tt::tt_metal::Tensor const&, tt::tt_metal::Device*, tt::tt_metal::MemoryConfig const&), tt::tt_metal::Tensor (*)(tt::tt_metal::Tensor const&, tt::tt_metal::Device*, tt::tt_metal::MemoryConfig const&)>::_M_invoke(std::_Any_data const&, tt::tt_metal::Tensor const&, tt::tt_metal::Device*&&, tt::tt_metal::MemoryConfig const&)
E        --- tt::tt_metal::tensor_impl::to_device_wrapper(tt::tt_metal::Tensor const&, tt::tt_metal::Device*, tt::tt_metal::MemoryConfig const&)
E        --- tt::tt_metal::Tensor::to(tt::tt_metal::Device*, tt::tt_metal::MemoryConfig const&) const
E        --- /home/ubuntu/uma/tt-metal/tt_eager/tt_lib/_C.so(+0x1f770b) [0x7f88f8d4370b]
E        --- /home/ubuntu/uma/tt-metal/tt_`

--- a/docs/aspell-dictionary.pws
+++ b/docs/aspell-dictionary.pws
@@ -299,6 +299,7 @@ intrinsics
 iomanip
 ios
 ipynb
+isclose
 isfinite
 isinf
 isnan

--- a/docs/aspell-dictionary.pws
+++ b/docs/aspell-dictionary.pws
@@ -174,6 +174,7 @@ args
 asin
 asinh
 asm
+assign
 async
 atan
 atanh

--- a/docs/source/ttnn/operations/binary.rst
+++ b/docs/source/ttnn/operations/binary.rst
@@ -7,6 +7,8 @@ Binary Operations
 
 .. autofunction:: ttnn.atan2
 
+.. autofunction:: ttnn.isclose
+
 .. autofunction:: ttnn.hypot
 
 .. autofunction:: ttnn.ldexp

--- a/docs/source/ttnn/operations/binary.rst
+++ b/docs/source/ttnn/operations/binary.rst
@@ -3,6 +3,8 @@ Binary Operations
 
 .. autofunction:: ttnn.addalpha
 
+.. autofunction:: ttnn.assign
+
 .. autofunction:: ttnn.atan2
 
 .. autofunction:: ttnn.hypot

--- a/docs/source/ttnn/operations/unary.rst
+++ b/docs/source/ttnn/operations/unary.rst
@@ -49,6 +49,8 @@ Unary Operations
 
 .. autofunction:: ttnn.hardtanh
 
+.. autofunction:: ttnn.heaviside
+
 .. autofunction:: ttnn.i0
 
 .. autofunction:: ttnn.isfinite

--- a/tests/ttnn/unit_tests/test_binary.py
+++ b/tests/ttnn/unit_tests/test_binary.py
@@ -17,7 +17,7 @@ from tests.ttnn.utils_for_testing import assert_with_pcc
     ),
 )
 class TestEltwiseBinary:
-    def test_ttnn_squared_difference(self, input_shape_a, input_shape_b, device):
+    def test_squared_difference(self, input_shape_a, input_shape_b, device):
         torch_a = torch.rand(input_shape_a, dtype=torch.bfloat16)
         torch_b = torch.rand(input_shape_b, dtype=torch.bfloat16)
 
@@ -36,7 +36,7 @@ class TestEltwiseBinary:
         assert_with_pcc(torch_output_tensor, output_tensor, 0.99)
 
     @pytest.mark.parametrize("alpha", [1.0, 2.5, -2.5])
-    def test_ttnn_subalpha(self, input_shape_a, input_shape_b, alpha, device):
+    def test_subalpha(self, input_shape_a, input_shape_b, alpha, device):
         torch.manual_seed(0)
 
         torch_a = torch.rand(input_shape_a, dtype=torch.bfloat16)
@@ -57,7 +57,7 @@ class TestEltwiseBinary:
         assert_with_pcc(torch_output_tensor, output_tensor, 0.99)
 
     @pytest.mark.parametrize("alpha", [1.0, 2.5, -2.5])
-    def test_ttnn_addalpha(self, input_shape_a, input_shape_b, alpha, device):
+    def test_addalpha(self, input_shape_a, input_shape_b, alpha, device):
         torch.manual_seed(0)
 
         torch_a = torch.rand(input_shape_a, dtype=torch.bfloat16)
@@ -77,7 +77,7 @@ class TestEltwiseBinary:
 
         assert_with_pcc(torch_output_tensor, output_tensor, 0.99)
 
-    def test_ttnn_atan2(self, input_shape_a, input_shape_b, device):
+    def test_atan2(self, input_shape_a, input_shape_b, device):
         torch_a = torch.rand(input_shape_a, dtype=torch.bfloat16)
         torch_b = torch.rand(input_shape_b, dtype=torch.bfloat16)
 
@@ -95,7 +95,7 @@ class TestEltwiseBinary:
 
         assert_with_pcc(torch_output_tensor, output_tensor, 0.99)
 
-    def test_ttnn_hypot(self, input_shape_a, input_shape_b, device):
+    def test_hypot(self, input_shape_a, input_shape_b, device):
         torch_a = torch.rand(input_shape_a, dtype=torch.bfloat16)
         torch_b = torch.rand(input_shape_b, dtype=torch.bfloat16)
 
@@ -113,7 +113,7 @@ class TestEltwiseBinary:
 
         assert_with_pcc(torch_output_tensor, output_tensor, 0.99)
 
-    def test_ttnn_ldexp(self, input_shape_a, input_shape_b, device):
+    def test_ldexp(self, input_shape_a, input_shape_b, device):
         torch_a = torch.rand(input_shape_a, dtype=torch.bfloat16)
         torch_b = torch.rand(input_shape_b, dtype=torch.bfloat16)
 
@@ -131,7 +131,7 @@ class TestEltwiseBinary:
 
         assert_with_pcc(torch_output_tensor, output_tensor, 0.99)
 
-    def test_ttnn_logaddexp(self, input_shape_a, input_shape_b, device):
+    def test_logaddexp(self, input_shape_a, input_shape_b, device):
         torch_a = torch.rand(input_shape_a, dtype=torch.bfloat16)
         torch_b = torch.rand(input_shape_b, dtype=torch.bfloat16)
 
@@ -149,7 +149,7 @@ class TestEltwiseBinary:
 
         assert_with_pcc(torch_output_tensor, output_tensor, 0.99)
 
-    def test_ttnn_logaddexp2(self, input_shape_a, input_shape_b, device):
+    def test_logaddexp2(self, input_shape_a, input_shape_b, device):
         torch_a = torch.rand(input_shape_a, dtype=torch.bfloat16)
         torch_b = torch.rand(input_shape_b, dtype=torch.bfloat16)
 
@@ -167,7 +167,7 @@ class TestEltwiseBinary:
 
         assert_with_pcc(torch_output_tensor, output_tensor, 0.98)
 
-    def test_ttnn_xlogy(self, input_shape_a, input_shape_b, device):
+    def test_xlogy(self, input_shape_a, input_shape_b, device):
         torch_a = torch.rand(input_shape_a, dtype=torch.bfloat16)
         torch_b = torch.rand(input_shape_b, dtype=torch.bfloat16)
 
@@ -185,7 +185,7 @@ class TestEltwiseBinary:
 
         assert_with_pcc(torch_output_tensor, output_tensor, 0.99)
 
-    def test_ttnn_logical_or(self, input_shape_a, input_shape_b, device):
+    def test_logical_or(self, input_shape_a, input_shape_b, device):
         torch_a = torch.rand(input_shape_a, dtype=torch.bfloat16)
         torch_b = torch.rand(input_shape_b, dtype=torch.bfloat16)
 
@@ -203,7 +203,7 @@ class TestEltwiseBinary:
 
         assert_with_pcc(torch_output_tensor, output_tensor, 0.99)
 
-    def test_ttnn_nextafter(self, input_shape_a, input_shape_b, device):
+    def test_nextafter(self, input_shape_a, input_shape_b, device):
         torch_a = torch.rand(input_shape_a, dtype=torch.bfloat16)
         torch_b = torch.rand(input_shape_b, dtype=torch.bfloat16)
 
@@ -221,7 +221,7 @@ class TestEltwiseBinary:
 
         assert_with_pcc(torch_output_tensor, output_tensor, 0.99)
 
-    def test_ttnn_logical_xor(self, input_shape_a, input_shape_b, device):
+    def test_logical_xor(self, input_shape_a, input_shape_b, device):
         torch_a = torch.rand(input_shape_a, dtype=torch.bfloat16)
         torch_b = torch.rand(input_shape_b, dtype=torch.bfloat16)
 
@@ -239,7 +239,7 @@ class TestEltwiseBinary:
 
         assert_with_pcc(torch_output_tensor, output_tensor, 0.99)
 
-    def test_ttnn_logical_and(self, input_shape_a, input_shape_b, device):
+    def test_logical_and(self, input_shape_a, input_shape_b, device):
         torch_a = torch.rand(input_shape_a, dtype=torch.bfloat16)
         torch_b = torch.rand(input_shape_b, dtype=torch.bfloat16)
 
@@ -256,3 +256,21 @@ class TestEltwiseBinary:
         output_tensor = ttnn.to_torch(output_tensor)
 
         assert_with_pcc(torch_output_tensor, output_tensor, 0.99)
+
+    def test_assign(self, input_shape_a, input_shape_b, device):
+        torch_a = torch.rand(input_shape_a, dtype=torch.bfloat16)
+        torch_b = torch.rand(input_shape_b, dtype=torch.bfloat16)
+
+        torch_b.copy_(torch_a)
+
+        a = ttnn.from_torch(torch_a)
+        b = ttnn.from_torch(torch_b)
+
+        a = ttnn.to_device(a, device)
+        b = ttnn.to_device(b, device)
+
+        output_tensor = ttnn.assign(a, b)
+        output_tensor = ttnn.to_layout(output_tensor, ttnn.ROW_MAJOR_LAYOUT)
+        output_tensor = ttnn.to_torch(output_tensor)
+
+        assert_with_pcc(torch_b, output_tensor, 0.99)

--- a/tests/ttnn/unit_tests/test_binary.py
+++ b/tests/ttnn/unit_tests/test_binary.py
@@ -274,3 +274,29 @@ class TestEltwiseBinary:
         output_tensor = ttnn.to_torch(output_tensor)
 
         assert_with_pcc(torch_b, output_tensor, 0.99)
+
+    @pytest.mark.parametrize(
+        "rtol, atol, equal_nan",
+        (
+            (1e-3, 1e-2, False),
+            (1e-5, 1e-4, True),
+            (1e-7, 1e-6, True),
+        ),
+    )
+    def test_isclose(self, input_shape_a, input_shape_b, rtol, atol, equal_nan, device):
+        torch_a = torch.rand(input_shape_a, dtype=torch.bfloat16)
+        torch_b = torch.rand(input_shape_b, dtype=torch.bfloat16)
+
+        torch_output_tensor = torch.isclose(torch_a, torch_b, rtol=rtol, atol=atol, equal_nan=equal_nan)
+
+        a = ttnn.from_torch(torch_a)
+        b = ttnn.from_torch(torch_b)
+
+        a = ttnn.to_device(a, device)
+        b = ttnn.to_device(b, device)
+
+        output_tensor = ttnn.isclose(a, b, rtol, atol, equal_nan)
+        output_tensor = ttnn.to_layout(output_tensor, ttnn.ROW_MAJOR_LAYOUT)
+        output_tensor = ttnn.to_torch(output_tensor)
+
+        assert_with_pcc(torch_output_tensor, output_tensor, 0.99)

--- a/ttnn/binary.py
+++ b/ttnn/binary.py
@@ -235,6 +235,84 @@ def register_ttl_assign_function(name, ttl_binary_function, torch_function):
     return binary_function
 
 
+def register_ttl_binary_function_with_multiple_parameter(name, ttl_binary_function, torch_function):
+    def _torch_binary(input_tensor_a: Tensor, input_tensor_b: Tensor, parameter_1, parameter_2, equal_nan, **_):
+        import torch
+        import ttnn
+
+        input_tensor_a = ttnn.from_device(input_tensor_a)
+        input_tensor_a = ttnn.to_layout(input_tensor_a, ttnn.ROW_MAJOR_LAYOUT)
+        input_tensor_a = ttnn.to_torch(input_tensor_a)
+
+        input_tensor_b = ttnn.from_device(input_tensor_b)
+        input_tensor_b = ttnn.to_layout(input_tensor_b, ttnn.ROW_MAJOR_LAYOUT)
+        input_tensor_b = ttnn.to_torch(input_tensor_b)
+        assert torch_function, f"Torch function not implemented for {str(ttl_binary_function)}"
+        return torch_function(input_tensor_a, input_tensor_b, parameter_1, parameter_2, equal_nan)
+
+    @decorate_operation(torch_function=_torch_binary, name=name)
+    def binary_function(
+        input_tensor_a: Tensor,
+        input_tensor_b: Tensor,
+        parameter_1: float,
+        parameter_2: float,
+        equal_nan: bool,
+        *,
+        memory_config: MemoryConfig = DRAM_MEMORY_CONFIG,
+    ) -> Tensor:
+        f"""{name}(input_tensor_a: Tensor, input_tensor_b: Tensor) -> Tensor
+
+        Applies {name} to :attr:`input_tensor_a`  and  :attr:`input_tensor_b` element-wise.
+
+        .. math::
+            {name}(\\mathrm{{input\\_tensor_a}}_i)
+            {name}(\\mathrm{{input\\_tensor_b}}_i)
+
+        Args:
+            * :attr:`input_tensor_a`
+            * :attr:`input_tensor_b`
+
+        Example::
+
+            >>> tensor_a = ttnn.to_device(ttnn.from_torch(torch.tensor((1, 2), dtype=torch.bfloat16)), device)
+            >>> tensor_b = ttnn.to_device(ttnn.from_torch(torch.tensor((2, 2), dtype=torch.bfloat16)), device)
+            >>> output = ttnn.{name}(tensor_a, tensor_b, 0.4 , 0.1)
+            >>> print(output)
+            Tensor([ -3, -2], dtype=bfloat16 )
+
+        """
+
+        if not (input_tensor_a.shape == input_tensor_b.shape):
+            raise RuntimeError("input_tensors must be of same size!")
+
+        original_shape = input_tensor_a.shape
+        input_tensor_a = _reshape_to_4D(input_tensor_a)
+        input_tensor_b = _reshape_to_4D(input_tensor_b)
+
+        if not isinstance(input_tensor_a, Tensor) or not isinstance(input_tensor_b, Tensor):
+            raise TypeError("Expected both arguments to be a ttnn.Tensor")
+
+        if not has_storage_type_of(input_tensor_a, DEVICE_STORAGE_TYPE) or not has_storage_type_of(
+            input_tensor_b, DEVICE_STORAGE_TYPE
+        ):
+            raise RuntimeError("input_tensors must be on device!")
+
+        ttl_input_tensor_a = input_tensor_a.value
+        ttl_input_tensor_b = input_tensor_b.value
+
+        ttl_output_tensor = ttl_binary_function(
+            ttl_input_tensor_a, ttl_input_tensor_b, parameter_1, parameter_2, equal_nan, output_mem_config=memory_config
+        )
+
+        output_tensor = Tensor(ttl_output_tensor)
+        output_tensor = reshape(output_tensor, original_shape)
+        return output_tensor
+
+    setattr(THIS_MODULE, name, binary_function)
+    __all__.append(name)
+    return binary_function
+
+
 # register functions
 
 
@@ -283,3 +361,11 @@ TTL_FUNCTION_ASSIGN = [
 
 for binary_function_name, ttl_binary_function, torch_function in TTL_FUNCTION_ASSIGN:
     register_ttl_assign_function(binary_function_name, ttl_binary_function, torch_function)
+
+
+TTL_FUNCTION_WITH_MULTIPLE_PARAMETER = [
+    ("isclose", ttl.tensor.isclose, torch.isclose),
+]
+
+for binary_function_name, ttl_binary_function, torch_function in TTL_FUNCTION_WITH_MULTIPLE_PARAMETER:
+    register_ttl_binary_function_with_multiple_parameter(binary_function_name, ttl_binary_function, torch_function)

--- a/ttnn/binary.py
+++ b/ttnn/binary.py
@@ -168,12 +168,84 @@ def register_ttl_binary_function_with_float_parameter(name, ttl_binary_function,
     return binary_function
 
 
+def register_ttl_assign_function(name, ttl_binary_function, torch_function):
+    def _torch_binary(input_tensor_a: Tensor, input_tensor_b: Tensor, **_):
+        import torch
+        import ttnn
+
+        input_tensor_a = ttnn.from_device(input_tensor_a)
+        input_tensor_a = ttnn.to_layout(input_tensor_a, ttnn.ROW_MAJOR_LAYOUT)
+        input_tensor_a = ttnn.to_torch(input_tensor_a)
+
+        input_tensor_b = ttnn.from_device(input_tensor_b)
+        input_tensor_b = ttnn.to_layout(input_tensor_b, ttnn.ROW_MAJOR_LAYOUT)
+        input_tensor_b = ttnn.to_torch(input_tensor_b)
+        assert torch_function, f"Torch function not implemented for {str(ttl_binary_function)}"
+        return torch_function(input_tensor_a, input_tensor_b)
+
+    @decorate_operation(torch_function=_torch_binary, name=name)
+    def binary_function(input_tensor_a: Tensor, input_tensor_b: Tensor) -> Tensor:
+        f"""{name}(input_tensor_a: Tensor, input_tensor_b: Tensor) -> Tensor
+
+        Applies {name} to :attr:`input_tensor_a` and  :attr:`input_tensor_b` element-wise.
+
+        .. math::
+            {name}(\\mathrm{{input\\_tensor_a}}_i)
+            {name}(\\mathrm{{input\\_tensor_b}}_i)
+
+        Args:
+            * :attr:`input_tensor_a`
+            * :attr:`input_tensor_b`
+
+        Example::
+
+            >>> tensor_a = ttnn.to_device(ttnn.from_torch(torch.tensor((1, 2), dtype=torch.bfloat16)), device)
+            >>> tensor_b = ttnn.to_device(ttnn.from_torch(torch.tensor((2, 2), dtype=torch.bfloat16)), device)
+            >>> output = ttnn.{name}(tensor_a, tensor_b)
+            >>> print(output)
+            Tensor([ 2, 2], dtype=bfloat16 )
+
+        """
+        if not (input_tensor_a.shape == input_tensor_b.shape):
+            raise RuntimeError("input_tensors must be of same size!")
+
+        original_shape = input_tensor_a.shape
+        input_tensor_a = _reshape_to_4D(input_tensor_a)
+        input_tensor_b = _reshape_to_4D(input_tensor_b)
+
+        if not isinstance(input_tensor_a, Tensor) or not isinstance(input_tensor_b, Tensor):
+            raise TypeError("Expected both arguments to be a ttnn.Tensor")
+
+        if not has_storage_type_of(input_tensor_a, DEVICE_STORAGE_TYPE) or not has_storage_type_of(
+            input_tensor_b, DEVICE_STORAGE_TYPE
+        ):
+            raise RuntimeError("input_tensors must be on device!")
+
+        ttl_input_tensor_a = input_tensor_a.value
+        ttl_input_tensor_b = input_tensor_b.value
+
+        ttl_output_tensor = ttl_binary_function(ttl_input_tensor_a, ttl_input_tensor_b)
+
+        output_tensor = Tensor(ttl_output_tensor)
+        output_tensor = reshape(output_tensor, original_shape)
+        return output_tensor
+
+    setattr(THIS_MODULE, name, binary_function)
+    __all__.append(name)
+    return binary_function
+
+
 # register functions
 
 
 def torch_squared_difference(x, y, *args, **kwargs):
     t_diff = torch.sub(x, y)
     return torch.square(t_diff)
+
+
+def torch_assign_binary(x, y, *args, **kwargs):
+    y.copy_(x)
+    return y
 
 
 TTL_BINARY_FUNCTIONS = [
@@ -203,3 +275,11 @@ TTL_BINARY_FUNCTIONS_WITH_FLOAT_PARAMETER = [
 
 for binary_function_name, ttl_binary_function, torch_function in TTL_BINARY_FUNCTIONS_WITH_FLOAT_PARAMETER:
     register_ttl_binary_function_with_float_parameter(binary_function_name, ttl_binary_function, torch_function)
+
+
+TTL_FUNCTION_ASSIGN = [
+    ("assign", ttl.tensor.assign, torch_assign_binary),
+]
+
+for binary_function_name, ttl_binary_function, torch_function in TTL_FUNCTION_ASSIGN:
+    register_ttl_assign_function(binary_function_name, ttl_binary_function, torch_function)

--- a/ttnn/unary.py
+++ b/ttnn/unary.py
@@ -344,6 +344,12 @@ def torch_rdiv(x, *args, **kwargs):
     return dim / x
 
 
+def torch_heaviside(x, *args, **kwargs):
+    value = kwargs.pop("scalar")
+    result = torch.heaviside(x, torch.tensor(value, dtype=x.dtype))
+    return result
+
+
 TTL_UNARY_FUNCTIONS_WITH_FLOAT_PARAMETER = [
     ("pow", ttl_pow, torch.pow),
     ("elu", ttl.tensor.elu, F.elu),
@@ -352,6 +358,7 @@ TTL_UNARY_FUNCTIONS_WITH_FLOAT_PARAMETER = [
     ("threshold", ttl.tensor.threshold, torch.threshold),
     ("leaky_relu", ttl.tensor.leaky_relu, F.leaky_relu),
     ("hardshrink", ttl.tensor.hardshrink, F.hardshrink),
+    ("heaviside", ttl.tensor.heaviside, torch_heaviside),
     ("prelu", ttl.tensor.prelu, torch.prelu),
     ("polygamma", ttl.tensor.polygamma, torch.polygamma),
     ("logit", ttl.tensor.logit, torch.logit),

